File: frontend/src/types/organization.ts
/**
 * Organization types based on backend schema
 */

export interface Organization {
  id: string;
  name: string;
  company_name: string | null;
  document: string | null;
  email: string | null;
  phone: string | null;
  address: string | null;
  logo_url: string | null;
  settings: Record<string, unknown>;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  owner_id: string | null;
  owner_email: string | null;
  owner_first_name: string | null;
  owner_last_name: string | null;
}

export interface OrganizationCreate {
  name: string;
  company_name?: string;
  document?: string;
  email?: string;
  phone?: string;
  address?: string;
  logo_url?: string;
  settings?: Record<string, unknown>;
  // Owner fields (required for creation)
  owner_email: string;
  owner_password: string;
  owner_first_name?: string;
  owner_last_name?: string;
}

export interface OrganizationUpdate {
  name?: string;
  company_name?: string;
  document?: string;
  email?: string;
  phone?: string;
  address?: string;
  logo_url?: string;
  settings?: Record<string, unknown>;
  is_active?: boolean;
  // Owner fields (for updating owner data)
  owner_first_name?: string;
  owner_last_name?: string;
  owner_password?: string; // Only if resetting password
}

File: backend/app/schemas/organization.py
"""Schemas de organização."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, EmailStr


class OrganizationBase(BaseModel):
    """Schema base de organização."""

    name: str
    company_name: str | None = None
    document: str | None = None
    email: EmailStr | None = None
    phone: str | None = None
    address: str | None = None


class OrganizationCreate(OrganizationBase):
    """Schema para criação de organização."""

    # Dados do owner inicial
    owner_email: EmailStr
    owner_password: str
    owner_first_name: str | None = None
    owner_last_name: str | None = None


class OrganizationUpdate(BaseModel):
    """Schema para atualização de organização."""

    name: str | None = None
    company_name: str | None = None
    document: str | None = None
    email: EmailStr | None = None
    phone: str | None = None
    address: str | None = None
    logo_url: str | None = None
    settings: dict | None = None
    is_active: bool | None = None


class OrganizationResponse(OrganizationBase):
    """Schema de resposta de organização."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    logo_url: str | None
    settings: dict
    is_active: bool
    created_at: datetime
    updated_at: datetime


class OrganizationWithOwner(OrganizationResponse):
    """Schema de organização com dados do owner."""

    owner_id: UUID | None = None
    owner_email: str | None = None
    owner_first_name: str | None = None
    owner_last_name: str | None = None